scripts/emulate_app.sh

```diff
@@ -0,0 +1,283 @@
+            echo -e "      \t\t  'armeabi-v7a' - default system image: API 30 google_apis_playstore x86"
+            echo -e "\n  -l\t\tSelect API level (optional)"
+            echo -e "      \t\t  Minimum API level: 23"
+            echo -e "\n  -t\t\tSelect API target (optional)"
+            echo -e "      \t\t  See examples on selecting system images below"
+            echo -e "\n  -s\t\tCreate an AVD and snapshot for quick-boot (optional)"
+            echo -e "      \t\t  Does not run app"
+            echo -e "\n  -x\t\tSet timeout in seconds for emulator launch and AVD boot-up (optional)"
+            echo -e "      \t\t  Default: 1800"
+            echo -e "      \t\t  Must be an integer"
+            echo -e "\nExamples:"
+            echo -e "  '$(basename $0) -a x86_64 -s'\tCreates an AVD and quick-boot snapshot for x86_64 ABI"
+            echo -e "  '$(basename $0) -a x86_64'   \tEmulates app for x86_64 ABI from snapshot"
+            echo -e "  '$(basename $0) -a x86 -l 29 -t google_apis'"
+            echo -e "                             \t\tSelect system image \"system-images;android-29;google_apis;x86\""
+            echo -e "\nRecommended system images for testing ARM ABIs:"
+            echo -e "  armeabi-v7a:"
+            echo -e "    \"system-images;android-30;google_apis_playstore;x86\" - default"
+            echo -e "    \"system-images;android-30;google-tv;x86\""
+            # TODO: add list of supported images for arm64-v8a
+            echo -e "\nFor a full list of system images run 'sdkmanager --list'"
+            exit 1
+            ;;
+        ?)
+            echo "Try '$(basename $0) -h' for more information." >&2
+            exit 1
+            ;;
+    esac
+done
+if [[ $set_abi == false ]]; then 
+    echo "Error: ABI not specified" >&2
+    echo "Try '$(basename $0) -h' for more information." >&2
+    exit 1
+fi
+
+# Set defaults
+if [[ $set_api_level == false ]]; then
+    api_level=$api_level_default
+fi
+if [[ $set_api_target == false ]]; then
+    api_target=$api_target_default
+fi
+
+# Setup working directory
+if [ ! -d "./android/app" ]; then
+    echo "Error: Incorrect working directory" >&2
+    echo "Try './scripts/$(basename $0)' from zingo-mobile root directory." >&2
+    exit 1
+fi
+cd android
+
+echo -e "\nInstalling latest build tools, platform tools, and platform..."
+sdkmanager --install 'build-tools;33.0.2' platform-tools
+
+echo "Installing latest emulator..."
+sdkmanager --install emulator --channel=0
+
+echo "Installing system image..."
+avd_name="${avd_skin}-android-${api_level}_${api_target}_${arch}"
+sdk="system-images;android-${api_level};${api_target};${arch}"
+sdkmanager --install "${sdk}"
+sdkmanager --licenses
+
+# Kill all emulators
+../scripts/kill_emulators.sh
+
+if [[ $create_snapshot == true ]]; then
+    echo -e "\nCreating AVD..."
+    echo no | avdmanager create avd --force --name "${avd_name}" --package "${sdk}" --device "${avd_skin}" --skin "${avd_skin}"
```

Comment:
the skin causes a problem for me locally. i dont have these skins downloaded

Reply:
oh i dont remember manually dowloading these, it must share a folder with android studio. i will try to add something to setup skins now